catroid/src/org/catrobat/catroid/content/bricks/PhiroIfLogicBeginBrick.java

```diff
@@ -0,0 +1,263 @@
+		ArrayAdapter<CharSequence> phiroProSensorAdapter = ArrayAdapter.createFromResource(context,
+				R.array.brick_phiro_select_sensor_spinner, android.R.layout.simple_spinner_item);
+		phiroProSensorAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+
+		if (checkbox.getVisibility() == View.VISIBLE) {
+			phiroProSensorSpinner.setClickable(false);
+			phiroProSensorSpinner.setEnabled(false);
+		} else {
+			phiroProSensorSpinner.setClickable(true);
+			phiroProSensorSpinner.setEnabled(true);
+			phiroProSensorSpinner.setOnItemSelectedListener(this);
+		}
+
+		phiroProSensorSpinner.setAdapter(phiroProSensorAdapter);
+		phiroProSensorSpinner.setSelection(sensorSpinnerPosition);
+
+		phiroProSensorSpinner.setOnItemSelectedListener(new OnItemSelectedListener() {
+
+			@Override
+			public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+				sensorSpinnerPosition = position;
+			}
+
+			@Override
+			public void onNothingSelected(AdapterView<?> arg0) {
+			}
+		});
+
+		return view;
+	}
+
+	@Override
+	public View getViewWithAlpha(int alphaValue) {
+		if (view != null) {
+			View layout = view.findViewById(R.id.brick_phiro_sensor_layout);
+			Drawable background = layout.getBackground();
+			background.setAlpha(alphaValue);
+
+			Spinner phiroProSensorSpinner = (Spinner) view.findViewById(R.id.brick_phiro_sensor_action_spinner);
+			phiroProSensorSpinner.getBackground().setAlpha(alphaValue);
+
+			TextView ifLabel = (TextView) view.findViewById(R.id.brick_phiro_sensor_label);
+			TextView ifLabelEnd = (TextView) view.findViewById(R.id.phiro_sensor_label_second_part);
+
+			ifLabel.setTextColor(ifLabel.getTextColors().withAlpha(alphaValue));
+			ifLabelEnd.setTextColor(ifLabelEnd.getTextColors().withAlpha(alphaValue));
+			this.alphaValue = (alphaValue);
+		}
+
+		return view;
+	}
+
+	@Override
+	public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+	}
+
+	@Override
+	public void onNothingSelected(AdapterView<?> arg0) {
+	}
+
+	@Override
+	public View getPrototypeView(Context context) {
+		View prototypeView = View.inflate(context, R.layout.brick_phiro_if_sensor, null);
+
+		Spinner phiroProSensorSpinner = (Spinner) prototypeView.findViewById(R.id.brick_phiro_sensor_action_spinner);
+		phiroProSensorSpinner.setFocusableInTouchMode(false);
+		phiroProSensorSpinner.setFocusable(false);
+
+		ArrayAdapter<CharSequence> phiroProSensorSpinnerAdapter = ArrayAdapter.createFromResource(context,
+				R.array.brick_phiro_select_sensor_spinner, android.R.layout.simple_spinner_item);
+		phiroProSensorSpinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+
+		phiroProSensorSpinner.setAdapter(phiroProSensorSpinnerAdapter);
+		phiroProSensorSpinner.setSelection(sensorSpinnerPosition);
+
+		return prototypeView;
+	}
+
+	@Override
+	public boolean isInitialized() {
+		if (ifElseBrick == null) {
+			return false;
+		} else {
+			return true;
+		}
```

Comment:
A checkstyle rule is waiting to handle such simplifications. :smirk_cat:

However, the class is a whole c&p of the base class IfLogicBeginBrick and many of these methods shouldn't be overwritten. Or at least not overwritten, if the code is exactly the same. Also some members don't have to be redeclared the same way as in the base class. (e.g. ifElseBrick or iEndBrick)